include/micm/cuda/solver/cuda_state.hpp

```diff
@@ -19,9 +19,21 @@ namespace micm
 
     CudaMatrixParam absolute_tolerance_param_;
 
+    // for NormalizedError function
+    std::size_t errors_size_;
+    double* errors_input_ = nullptr;
+    double* errors_output_ = nullptr;
+
+    // for AlphaMinusJacobian function
+    std::size_t jacobian_diagonal_elements_size_;
+    std::size_t* jacobian_diagonal_elements_ = nullptr;
```

Comment:
so maybe make these as one struct so that we can pass the struct around rather than the raw pointers and size?